plugins/modules/ec2_snapshot.py

```diff
@@ -163,13 +164,14 @@ def _get_most_recent_snapshot(snapshots, max_snapshot_age_secs=None, now=None):
         return None
 
     if not now:
-        now = datetime.datetime.utcnow()
-
-    youngest_snapshot = max(snapshots, key=_get_snapshot_starttime)
+        now = datetime.datetime.now(datetime.timezone.utc)
 
-    # See if the snapshot is younger that the given max age
-    snapshot_start = datetime.datetime.strptime(youngest_snapshot.start_time, '%Y-%m-%dT%H:%M:%S.%fZ')
-    snapshot_age = now - snapshot_start
+    youngest_snapshot = max(snapshots, key=lambda s: s['StartTime'])
+    snapshot_start = youngest_snapshot['StartTime']
+    try:
+        snapshot_age = now - snapshot_start
+    except Exception as e:
+        module.fail_json_aws(e, msg="bang", now=now, snapshot_start=snapshot_start)
```

Comment:
Is "bang" just here for debugging?

Reply:
Yeah.  I was having fun with "offset aware/naïve" datetime objects.

If you see fail/exceptions with "bang"/boom" it's probably debugging that's snuck through.  Similarly catching Exception is usually me being lazy during debugging.